src/storage/storage_structure/storage_structure.cpp

```diff
@@ -16,12 +16,14 @@ static inline bool isInRange(uint64_t val, uint64_t start, uint64_t end) {
 void StorageStructure::addNewPageToFileHandle() {
     auto pageIdxInOriginalFile = fileHandle->addNewPage();
     auto pageIdxInWAL = wal->logPageInsertRecord(storageStructureID, pageIdxInOriginalFile);
-    bufferManager.pinWithoutAcquiringPageLock(
+    bufferManager.pin(
         *wal->fileHandle, pageIdxInWAL, BufferManager::PageReadPolicy::DONT_READ_PAGE);
-    fileHandle->createPageVersionGroupIfNecessary(pageIdxInOriginalFile);
-    fileHandle->setWALPageVersion(pageIdxInOriginalFile, pageIdxInWAL);
+    fileHandle->addWALPageIdxGroupIfNecessary(pageIdxInOriginalFile);
+    fileHandle->setWALPageIdx(pageIdxInOriginalFile, pageIdxInWAL);
+    // TODO(Reviewer): Do we need to set the page dirty here? Instead, can we only set the page
```

Comment:
If you have added a brand new page to a file, that is a write operation, so that page needs to be flushed so that it persists if it gets evicted. So we should mark the page as dirty.